
# 20250627, Adriaan van Os <adriaan@adriaan.biz>
# Work-around for fpc issue #40628 which makes macpas mode unuseable
# <https://gitlab.com/freepascal.org/fpc/source/-/issues/40628>
# note that SetExceptionMask must instead be called at the start of application code

--- a/rtl/inc/macpas.pp	2026-01-06 16:29:04
+++ b/rtl/inc/macpas.pp	2026-01-07 17:14:04
@@ -109,14 +109,6 @@
 
 implementation
 
-{$IFDEF FPC_DOTTEDUNITS}
-uses
-  System.Math;
-{$ELSE}
-uses
-  math;
-{$ENDIF}
-
 {$r-}
 {$q-}
 
@@ -398,8 +390,5 @@
 end;
 
 
-{$ifndef FPUNONE}
 begin
-  SetExceptionMask([exInvalidOp, exDenormalized, exZeroDivide, exOverflow, exUnderflow, exPrecision])
-{$endif}
 end.
